recipe-client-addon/test/unit/test_SandboxManager.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+Cu.import("resource://shield-recipe-client/lib/SandboxManager.jsm");
+
+// wrapAsync should wrap privileged Promises with Promises that are usable by
+// the sandbox.
+add_task(function* () {
+  const manager = new SandboxManager();
+  manager.addHold("testing");
+
+  manager.cloneIntoGlobal("driver", {
+    async privileged() {
+      return "privileged";
+    },
+    wrapped: manager.wrapAsync(async function() {
+      return "wrapped";
+    }),
+  }, {cloneFunctions: true});
+
+  // Assertion helpers
+  manager.addGlobal("ok", ok);
+
+  const wrappedResult = yield new Promise(resolve => {
+    manager.addGlobal("resolve", result => resolve(result));
+    manager.evalInSandbox(`
+      // Unwrapped privileged promises are not accessible in the sandbox
+      try {
+        const privilegedResult = driver.privileged().then(() => false);
+        ok(false, "The sandbox could not use a privileged Promise");
+      } catch (err) { }
+
+      // Wrapped functions return promises that the sandbox can access.
+      const wrappedResult = driver.wrapped();
```

Comment:
It is confusing that both this value and the one on line 23 are both called `wrappedResult`. I tried for a while to see how they were related. Can one of them be a different name?